project_recalculate/models/project_task.py

```diff
@@ -235,23 +249,29 @@ def task_recalculate(self):
             date_end = False
             from_days = self._from_days_dec(
                 from_days, project_date, resource, calendar, increment)
-            start = self._calendar_schedule_days(
-                from_days, project_date, resource, calendar)[1]
-            if start:
-                day = start[0].replace(hour=0, minute=0, second=0)
+            planned_dt = self._calendar_plan_days(
+                from_days, project_date, resource, calendar)
+            if planned_dt:
+                day = planned_dt.replace(hour=0, minute=0, second=0)
                 first = self._first_interval_of_day_get(
                     day, resource, calendar)
                 if first:
                     date_start = first[0]
             if date_start:
-                end = self._calendar_schedule_days(
-                    task.estimated_days, date_start, resource, calendar)[1]
-                if end:
-                    date_end = end[1]
+                end_planned_dt = self._calendar_plan_days(
+                    task.estimated_days, date_start, resource, calendar)
+                if end_planned_dt:
+                    date_end = self._last_interval_of_day_get(
+                        end_planned_dt, resource, calendar)[1]
+            if date_start:
+                date_start = date_start.astimezone(utc)
+            if date_end:
+                date_end = date_end.astimezone(utc)
+
             task.with_context(task.env.context, task_recalculate=True).write({
-                'date_start': date_start and to_string(date_start) or False,
-                'date_end': date_end and to_string(date_end) or False,
-                'date_deadline': date_end and to_string(date_end) or False,
+                'date_start': date_start and date_start or False,
```

Comment:
Oops, did you mean this?

```suggestion
                'date_start': date_start or False,
```